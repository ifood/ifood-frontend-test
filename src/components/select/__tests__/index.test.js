import React from 'react'
import { useSelector } from 'react-redux'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'

import SelectProvider, { Select } from '../index'

afterEach(cleanup)

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}))

describe('<Select />', () => {
  beforeEach(() => {
    useSelector.mockImplementation((cb) =>
      cb({
        filter: {
          filters: [
            {
              id: 'limit',
              name: 'Quantidade',
              validation: { min: 1, max: 50 },
            },
          ],
          currentFilters: {
            name: '',
            country: '',
            locale: '',
            limit: 20,
            offset: 1,
            date: null,
          },
        },
        playlist: {
          playlists: [],
          total: 20,
        },
      })
    )
  })

  afterEach(() => {
    useSelector.mockClear()
  })

  it('Should be able to render without errors', () => {
    const { getByText } = render(<SelectProvider id='limit' />)

    const title = getByText('Quantidade')

    expect(title).toBeTruthy()
  })

  it('Should be able to render without errors', () => {
    const mockedOptions = [5, 10, 15, 20, 30]
    const { asFragment } = render(<Select options={mockedOptions} value='20' />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('Should be able to render limit filter', async () => {
    const mockedOptions = ['5', '10', '15', '20']
    // eslint-disable-next-line prettier/prettier
    const { getByText, queryByTestId } = render(<Select options={mockedOptions} value='5' />)

    const selectContainer = queryByTestId('select-container')

    expect(selectContainer).toBeDefined()
    expect(selectContainer).not.toBeNull()

    expect(getByText('5')).toBeTruthy()
    expect(getByText('15')).toBeTruthy()
  })
})
