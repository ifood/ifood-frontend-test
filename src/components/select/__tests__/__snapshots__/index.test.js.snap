// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Should be able to render without errors 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx eEAbop"
  >
    <span
      class="sc-gtssRu wOEbu"
    />
    <select
      class="sc-dlnjPT kSoqVB"
      data-testid="select-container"
    >
      <option
        class="sc-hKFyIo jWrmIL"
        value="5"
      >
        5
      </option>
      <option
        class="sc-hKFyIo jWrmIL"
        value="10"
      >
        10
      </option>
      <option
        class="sc-hKFyIo jWrmIL"
        value="15"
      >
        15
      </option>
      <option
        class="sc-hKFyIo jWrmIL"
        value="20"
      >
        20
      </option>
      <option
        class="sc-hKFyIo jWrmIL"
        value="30"
      >
        30
      </option>
    </select>
  </div>
</DocumentFragment>
`;
